### PR TITLE
 CreateExtensionMojo should optionally add the newly created modules to BOMs

### PR DESCRIPTION
**Pull request by ppalaga**
A followup of https://github.com/quarkusio/quarkus/pull/3390

### CODE REVIEW

**Comment by ppalaga** [comment]
@gsmet @aloubyansky or anybody else could you please review?

File: devtools/common/src/main/java/io/quarkus/maven/utilities/PomTransformer.java [hunk 1]
```diff
@@ -266,25 +286,61 @@ public static Transformation addModule(String module) {
...
+        public static Transformation addManagedDependency(String groupId, String artifactId, String version) {
+            return (Document document, TransformationContext context) -> {
+                try {
+                    Node dependencyManagementDeps = (Node) context.getXPath().evaluate(
+                            anyNs("project", "dependencyManagement", "dependencies"), document,
+                            XPathConstants.NODE);
+                    if (dependencyManagementDeps == null) {
+                        throw new IllegalStateException(
+                                String.format("//project/dependencyManagement/dependencies not found in [%s]",
+                                        context.getPomXmlPath()));
```
Comment (aloubyansky) [inline devtools/common/src/main/java/io/quarkus/maven/utilities/PomTransformer.java]: You didn't want to add it in case it's missing?

Comment (ppalaga) [reply devtools/common/src/main/java/io/quarkus/maven/utilities/PomTransformer.java]: I think if it is not there it rather means that the user has misconfigured the path.

Comment (ppalaga) [reply devtools/common/src/main/java/io/quarkus/maven/utilities/PomTransformer.java]: Thinking of it again, let me add it.

Comment (ppalaga) [reply devtools/common/src/main/java/io/quarkus/maven/utilities/PomTransformer.java]: Solved in b990aa5

**Comment by ppalaga** [comment]
In b990aa5 the `dependencyManagement/dependencies` elements are added to the BOM if necessary thus addressing the request of @aloubyansky.

**Review by aloubyansky — APPROVED** [review]

**Comment by ppalaga** [comment]
Is this good to merge?

**Pull request merged**